src/ducks/userInfo/userInfoReducer.jsx

```diff
@@ -24,6 +24,26 @@ const userInfoReducer = (state = {}, action) => {
             };
         case USER_REQUEST_ERROR:
             return {errorMessage: action.payload.message};
+        case USER_TASK_UPDATED: 
+            
+            const userWeek = {...state.userWeek};
+            const userData = {...state.userData};
+
+            userData.balance = action.payload.updatedBalance;
+            userWeek.rewardsGained = action.payload.updatedBalance;
+            
+            userWeek.tasks.forEach((task) => {
+                if (task._id === action.payload.updatedTask.id) {
+                    userWeek.tasks[userWeek.tasks.indexOf(task)].days = action.payload.updatedTask.days;
```

Comment:
Here you mutate store as tasks is array and copies by link, its better to use filter